crates/trie/src/prefix_set/mod.rs

```diff
@@ -121,6 +121,9 @@ impl PrefixSetMut {
             self.keys.dedup();
         }
 
+        // we need to shrink in both the sorted and non-sorted cases because deduping may have
+        // occurred either on `freeze`, or during `contains`.
+        self.keys.shrink_to_fit();
```

Comment:
what if we extract this to a helper function that we can reuse here and in contains?
setting the sorted flag on freeze when we don't need it shouldn't be a big deal